servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SingleConcatWithPublisher.java

```diff
@@ -145,24 +190,136 @@ public void request(long n) {
                 }
             }
         }
+    }
+
+    private static final class ConcatDeferNextSubscriber<T> extends AbstractConcatSubscriber<T> {
+        /**
+         * If only one item was {@link #request(long) requested} before {@link #onSuccess(Object)}.
+         */
+        private static final Object REQUESTED_ONE = new Object();
```

Comment:
I like defining toString for these to the name of the field. Some people use an enum if they have a bunch of them to define as the toString is free. Instance variables can use either the enum type or Object and if the enum type, it is nice to be able to use switch.

Reply:
I like an idea for using an `enum` in general, but it didn't fit well for this use-case because:
- set of states is different for each subscriber impl
- `mayBeResult` may also result of the `Single`
I will keep it consistent with all other operators and we can refactor them altogether if necessary